src/tep_sup.erl

```diff
@@ -6,14 +6,21 @@
 
 -behaviour(supervisor).
 
--export([start_link/0]).
+-export([start_link/1]).
 
 -export([init/1]).
 
 -define(SERVER, ?MODULE).
 
-start_link() ->
-    supervisor:start_link({local, ?SERVER}, ?MODULE, []).
+-spec start_link(string() | no_path) ->
+    {ok, pid()} |
+    ignore |
+    {error,
+        {already_started, pid()} |
+        {shutdown, term()} |
+        term()}.
```

Comment:
Wait, we didn't have this? Did we miss enforcing `spec`s on every function? Because I kinda wanna do that now.

Reply:
Same here. I think it should be necessary. Please create an issue fo that.